File: app/src/main/java/com/airbnb/ui/HeroItemLayout.java
package com.airbnb.ui;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import com.airbnb.R;
import com.airbnb.net.HeroItem;
import com.squareup.picasso.Picasso;

import butterknife.ButterKnife;
import butterknife.InjectView;

public class HeroItemLayout extends FrameLayout {

    @InjectView(R.id.title) public TextView mTitleView;
    @InjectView(R.id.text) public TextView mTextView;
    @InjectView(R.id.background_image) public ImageView mBackgroundImageView;

    public HeroItemLayout(Context context) {
        this(context, null);
    }

    public HeroItemLayout(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public HeroItemLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        LayoutInflater inflater = LayoutInflater.from(context);
        inflater.inflate(R.layout.hero_item_contents, this, true);
        ButterKnife.inject(this, this);
    }

    @Override
    protected void onDetachedFromWindow() {
        super.onDetachedFromWindow();
        Picasso.with(getContext()).cancelRequest(mBackgroundImageView);
    }

    public void setHeroItem(HeroItem heroItem) {
        mTitleView.setText(heroItem.title);
        mTextView.setText(heroItem.text);
        if (heroItem.textBackgroundColor == 0) {
            mTextView.setBackgroundColor(0);
        } else {
            mTextView.setBackgroundColor(heroItem.textBackgroundColor);
        }
        ImageView imageView = mBackgroundImageView;
        if (heroItem.imageUri != null) {
            Picasso.with(getContext())
                    .load(heroItem.imageUri)
                    .placeholder(R.color.hero_placeholder)
                    .fit()
                    .centerCrop()
                    .into(imageView);
        } else {
            imageView.setBackgroundResource(R.color.hero_placeholder);
        }
    }
}

File: app/src/main/java/com/airbnb/ui/HeroItemLayout.kt
package com.airbnb.ui

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import butterknife.InjectView

import com.airbnb.R
import com.airbnb.net.HeroItem
import com.squareup.picasso.Picasso

public class HeroItemLayout @jvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : FrameLayout(context, attrs, defStyleAttr) {

    public var mTitleView: TextView
    public var mTextView: TextView
    public var mBackgroundImageView: ImageView

    init {
        val inflater = LayoutInflater.from(context)
        inflater.inflate(R.layout.hero_item_contents, this, true)
        mTitleView = findViewById(R.id.title) as TextView
        mTextView = findViewById(R.id.text) as TextView
        mBackgroundImageView = findViewById(R.id.background_image) as ImageView
    }

    override fun onDetachedFromWindow() {
        super.onDetachedFromWindow()
        Picasso.with(getContext()).cancelRequest(mBackgroundImageView)
    }

    public fun setHeroItem(heroItem: HeroItem) {
        mTitleView.setText(heroItem.title)
        mTextView.setText(heroItem.text)
        mTextView.setBackgroundColor(heroItem.textBackgroundColor)

        if (heroItem.imageUri != null) {
            Picasso.with(getContext())
                    .load(heroItem.imageUri)
                    .placeholder(R.color.background_light_grey)
                    .fit()
                    .centerCrop()
                    .into(mBackgroundImageView)
        } else {
            mBackgroundImageView.setBackgroundResource(R.color.background_light_grey)
        }
    }
}
